lib/ably/rest/client.rb

```diff
@@ -434,11 +439,22 @@ def send_request(method, path, params, options)
         max_retry_duration = http_defaults.fetch(:max_retry_duration)
         requested_at       = Time.now
         retry_count        = 0
+        request_id         = nil
+        if add_request_ids
+          params = {} if params.nil?
```

Comment:
This is a tiny comment, but it seems unnecesary to dup an empty hash, so perhaps instead just:

```ruby
params = if params.nil?
  {}
else
  params.dup
end
```